layers/meta-balena-beaglebone/recipes-core/images/balena-image.inc

```diff
@@ -1,6 +1,11 @@
 IMAGE_FSTYPES:append:beaglebone = " balenaos-img balenaos-img.bmap"
 IMAGE_FSTYPES:append:beaglebone-pocket = " balenaos-img"
 IMAGE_FSTYPES:append:beagleboard-xm = " balenaos-img"
+IMAGE_FSTYPES:append:beagleplay = " balenaos-img"
+IMAGE_FSTYPES:append:beagleplay-k3r5 = " balenaos-img"
+
+BALENA_BOOT_SIZE:beagleplay="327680"
```

Comment:
Why use such a large partition? What is the actual size of the files in the boot partition?

Reply:
The `u-boot-bb.org` partition is reproduced below from my build machine. Looks like the total size of the files here is 672K, so we should be ok using the default. Will change and retest.

```
ubuntu@ip-172-31-16-228:~/balena-beaglebone/build/tmp/work/beagleplay_k3r5-poky-eabi/u-boot-bb.org/1_2024.04+gitAUTOINC+881379e9e2-r0/deploy-u-boot-bb.org$ ls -ltrh
total 668K
-rw-r--r-- 2 ubuntu ubuntu 127K May 17 15:56 u-boot-r5spl-beagleplay-2024.04+gitAUTOINC+881379e9e2-r0.bin
lrwxrwxrwx 2 ubuntu ubuntu   60 May 17 15:56 u-boot-r5spl.bin -> u-boot-r5spl-beagleplay-2024.04+gitAUTOINC+881379e9e2-r0.bin
lrwxrwxrwx 2 ubuntu ubuntu   60 May 17 15:56 u-boot-r5spl-beagleplay.bin -> u-boot-r5spl-beagleplay-2024.04+gitAUTOINC+881379e9e2-r0.bin
-rw-r--r-- 2 ubuntu ubuntu 260K May 17 15:56 tiboot3-am62x-gp-evm-beagleplay-k3r5-2024.04+gitAUTOINC+881379e9e2-r0.bin
lrwxrwxrwx 2 ubuntu ubuntu   73 May 17 15:56 tiboot3.bin -> tiboot3-am62x-gp-evm-beagleplay-k3r5-2024.04+gitAUTOINC+881379e9e2-r0.bin
lrwxrwxrwx 2 ubuntu ubuntu   73 May 17 15:56 tiboot3-am62x-gp-evm -> tiboot3-am62x-gp-evm-beagleplay-k3r5-2024.04+gitAUTOINC+881379e9e2-r0.bin
-rw-r--r-- 2 ubuntu ubuntu  194 May 17 15:56 extlinux.conf-beagleplay-k3r5-r0
lrwxrwxrwx 2 ubuntu ubuntu   32 May 17 15:56 extlinux.conf-beagleplay-k3r5 -> extlinux.conf-beagleplay-k3r5-r0
-rw-r--r-- 2 ubuntu ubuntu 260K May 17 15:56 tiboot3-am62x-gp-evm.bin
lrwxrwxrwx 1 ubuntu ubuntu   32 May 17 15:56 extlinux.conf -> extlinux.conf-beagleplay-k3r5-r0
```

Reply:
@floion - I was unable to remove this line. The `do_resin_boot_dirgen_and_deploy` task was failing (see snippet below):

```
ERROR: balena-image-flasher-1.0-r0 do_resin_boot_dirgen_and_deploy: resin-boot: Not enough space for atomic copy operations.
ERROR: balena-image-flasher-1.0-r0 do_resin_boot_dirgen_and_deploy: ExecutionError('/home/ubuntu/balena-beaglebone/build/tmp/work/beagleplay-poky-linux/balena-image-flasher/1.0-r0/temp/run.do_resin_boot_dirgen_and_deploy.2753352', 1, None, None)
ERROR: Logfile of failure stored in: /home/ubuntu/balena-beaglebone/build/tmp/work/beagleplay-poky-linux/balena-image-flasher/1.0-r0/temp/log.do_resin_boot_dirgen_and_deploy.2753352
ERROR: Task (/home/ubuntu/balena-beaglebone/build/../layers/meta-balena/meta-balena-common/recipes-core/images/balena-image-flasher.bb:do_resin_boot_dirgen_and_deploy) failed with exit code '1'
```

Keeping it in the recipe (for now)

Reply:
Hey @floion : Here is the output of my `ls -ltrh` command from my latest build directory:

```
ubuntu@ip-172-31-16-228:~/balena-beaglebone/build/tmp/work/beagleplay-poky-linux/balena-image/1.0-r0/resin-boot$ ls -ltrh
total 62M
-rw-r--r-- 1 ubuntu ubuntu 1.3M Jun  4 22:17 tispl.bin
-rw-r--r-- 1 ubuntu ubuntu 746K Jun  4 22:17 u-boot.img
-rw-r--r-- 1 ubuntu ubuntu 260K Jun  4 22:17 tiboot3.bin
-rw-r--r-- 1 ubuntu ubuntu  61M Jun  4 22:17 Image
-rw-r--r-- 1 ubuntu ubuntu  63K Jun  4 22:17 k3-am625-beagleplay.dtb
drwxr-xr-x 2 ubuntu ubuntu 4.0K Jun  4 22:17 extlinux
drwxr-xr-x 2 ubuntu ubuntu 4.0K Jun  4 22:17 splash
-rw-r--r-- 1 ubuntu ubuntu  187 Jun  4 22:17 os-release
drwxr-xr-x 3 ubuntu ubuntu 4.0K Jun  4 22:17 EFI
-rw-r--r-- 1 ubuntu ubuntu 2.8K Jun  4 22:17 device-type.json
drwxr-xr-x 2 ubuntu ubuntu 4.0K Jun  4 22:17 system-connections
drwxr-xr-x 2 ubuntu ubuntu 4.0K Jun  4 22:17 system-proxy
-rw-r--r-- 1 ubuntu ubuntu    0 Jun  4 22:17 extra_uEnv.txt
-rw-r--r-- 1 ubuntu ubuntu   39 Jun  4 22:17 image-version-info
-rw-r--r-- 1 ubuntu ubuntu   24 Jun  4 22:17 balena-image
-rw-r--r-- 1 ubuntu ubuntu   69 Jun  4 22:17 config.json
-rw-r--r-- 1 ubuntu ubuntu  919 Jun  4 22:17 balenaos.fingerprint
```

Looks like my build is storing the actual kernel image inside the resin_boot directory. Am assuming this is wrong ? How do I fix this ?

Reply:
I think one way to do this may be use a compressed version of the kernel image. Can you suggest how this may be done ?

Reply:
I assume you also have the kernel in the rootfs in the /boot directory. You should instruct u-boot to load the kernel from there instead and doing so you can remove the kernel from the boot partition.
Also, is there a need for `k3-am625-beagleplay.dtb` in the boot partition? I think you can delete this one also.

Reply:
As suggested by @floion , I changed the extlinux configuration to pick up the image and DTB files from the /boot partition. This reduced the size of the `resin-boot` partition to about 2.3 M (which enabled the removal of the extra `BALENA_BOOT_SIZE` line from this file).

Please review and resolve as needed.